Lib/test/test_shutil.py

```diff
@@ -1848,6 +1848,24 @@ def test_pathext(self):
             rv = shutil.which(program, path=self.temp_dir)
             self.assertEqual(rv, temp_filexyz.name)
 
+    # Issue 40592: See https://bugs.python.org/issue40592
+    @unittest.skipUnless(sys.platform == "win32", 'test specific to Windows')
+    def test_pathext_with_empty_str(self):
+        ext = ".xyz"
+        temp_filexyz = tempfile.NamedTemporaryFile(dir=self.temp_dir,
+                                                   prefix="Tmp2", suffix=ext)
+        os.chmod(temp_filexyz.name, stat.S_IXUSR | stat.S_IWUSR)
```

Comment:
This has no impact on Windows (except maybe clearing the read-only flag, which won't have any impact on this test anyway)